src/build/build_step.go

```diff
@@ -301,12 +295,19 @@ func buildTarget(tid int, state *core.BuildState, target *core.BuildTarget, runR
 	}
 
 	metadata.EndTime = time.Now()
+
 	checkLicences(state, target)
 
 	if runRemotely {
 		target.SetState(core.BuiltRemotely)
 		state.LogBuildResult(tid, target.Label, core.TargetBuilt, "Built remotely")
 		return nil
+	} else if target.BuildCouldModifyTarget() {
+		if err := storeTargetMetadata(target, metadata); err != nil {
+			// TODO(jpoole): the plz-out dir doesn't seem to exist at this point when it's run remotely so storing the md
```

Comment:
As discussed, this is correct... remote runs don't create any of the intermediate outputs. Need to make sure rex keeps working.